.github/workflows/ci-docker.yml

```diff
@@ -0,0 +1,48 @@
+name: pressio-containers
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'develop'
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    paths:
+      - 'docker_scripts/**'
+  schedule:
+    - cron:  '0 5 * * FRI'
+
+jobs:
+  CI:
+    strategy:
+      matrix:
+        config:
+          - {dockerfile: 'fedora-clang',         tag: 'latest'}
+          - {dockerfile: 'fedora-gnu',           tag: 'latest'}
+          # - {dockerfile: 'intel-oneapi',         tag: 'latest'}
+          - {dockerfile: 'ubuntu-clang',         tag: 'latest'}
+          - {dockerfile: 'ubuntu-gnu-trilinos',  tag: 'latest'}
+          - {dockerfile: 'ubuntu-gnu',           tag: 'latest'}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout out code
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v3
+        if: github.repository == 'Pressio/pressio' && github.ref == 'refs/heads/develop'
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and Push Docker images for Github Container Registry
+        uses: docker/build-push-action@v4
+        with:
+          tags: ghcr.io/pressio/${{ matrix.config.dockerfile }}:${{ matrix.config.tag }}
+          file: docker_scripts/${{ matrix.config.dockerfile }}.dockerfile
+          pull: true
+          push: ${{ github.repository == 'Pressio/pressio' && github.ref == 'refs/heads/develop' }}
```

Comment:
what does this workflow file do? (sorry for my lack of knowledge of this)

Reply:
This:
- builds the docker images (whenever changes in `docker_scripts` are detected AND once a week on Friday)
- pushes the built images when the job was triggered from `develop` branch (after PR gets merged or maybe after direct push)